### PR TITLE
Fix creditmemo tax calculation bug when using multiple discounts

### PR DESCRIPTION
**Pull request by seansan**
Fix creditmemo tax calculation bug when using multiple discounts

Needs code review and additional testing

But this has been tested in production and solved our errors with creditmemo calculation missing value

### CODE REVIEW

**Comment by LeeSaferite** [comment]
Can you give a detail description of the problem and how this code solves the issue?
Looking at the code I see what appears to be dead code. The changes in the collectTotals method seem to just be removing the 'tax' total collector.

**Comment by seansan** [comment]
When adding several product types and utilizing several (more than 1) discount code the credit memo calculation is not correct. 

Before the fix the order value is 184,40 and credit memo is 181,41
After the the fix the order value is 184,40 and credit memo is 184,40

So in essence it seems to solves an tax issue when utilizing cart discounts

**Comment by colinmollenhour** [comment]
Does this merge cleanly with 1.9.3.0?

**Comment by tmotyl** [comment]
I've changed the base to 1.9.3.x, however a confirmation is needed whether the bug occures in 1.9.3 still.

File: app/code/core/Mage/Sales/Model/Order/Creditmemo.php [hunk 1]
```diff
@@ -324,9 +324,25 @@ public function addItem(Mage_Sales_Model_Order_Creditmemo_Item $item)
      *
      * @return Mage_Sales_Model_Order_Creditmemo
      */
-    public function collectTotals()
-    {
-        foreach ($this->getConfig()->getTotalModels() as $model) {
+     public function collectTotals()
```
Comment (tmotyl) [inline app/code/core/Mage/Sales/Model/Order/Creditmemo.php]: whitespace issue here (space)

File: app/code/core/Mage/Sales/Model/Order/Creditmemo.php [hunk 2]
```diff
@@ -324,9 +324,25 @@ public function addItem(Mage_Sales_Model_Order_Creditmemo_Item $item)
...
+    {
+        $models = $this->getConfig()->getTotalModels();
+        $res = array();
+        $i=0;
```
Comment (tmotyl) [inline app/code/core/Mage/Sales/Model/Order/Creditmemo.php]: whitespaces

File: app/code/core/Mage/Sales/Model/Order/Creditmemo.php [hunk 3]
```diff
@@ -324,9 +324,25 @@ public function addItem(Mage_Sales_Model_Order_Creditmemo_Item $item)
...
+        $discount = 0;
+        foreach ($models as $name =>$model) {
+            if ($name=='discount') $discount = $i;
```
Comment (tmotyl) [inline app/code/core/Mage/Sales/Model/Order/Creditmemo.php]: whitespaces around == needed, and please wrap `$discount = $i;` with brackets `{}`

File: app/code/core/Mage/Sales/Model/Order/Creditmemo.php [hunk 4]
```diff
@@ -324,9 +324,25 @@ public function addItem(Mage_Sales_Model_Order_Creditmemo_Item $item)
...
+            if ($name=='tax') {
```
Comment (tmotyl) [inline app/code/core/Mage/Sales/Model/Order/Creditmemo.php]: same here

**Comment by tmotyl** [comment]
@seansan can you improve the pull request and confirm that the issue still exists?

**Comment by seansan** [comment]
Feel free to change pull. But more importantly there needs to be more than 1 test ....

**Comment by tmotyl** [comment]
@seansan I have no access to this branch, probably you forgot to select a checkbox like "allow contributors to edit..." when making pull request

**Pull request closed without merging**